## Experiment 1(a): Applying Zhu et al.'s model to various benchmark datasets on our system and comparing the results with the ones in the paper

In [2]:
import pandas as pd
from evaluator import ReflectionEvaluator

In [3]:
# 1. Map out your server's dataset structure
base_path = '/storage/hdd1/data/amanr.mds2024/robustsirr_test_dataset/'
output_path = './results/all_datasets_eval/'

datasets_to_run = {
    'nature20': 'nature20',
    'real20': 'real20',
    'OpenRR100': 'OpenRR100',
    'SIR2_Postcard': 'SIR2/PostcardDataset',
    'SIR2_SolidObject': 'SIR2/SolidObjectDataset',
    'SIR2_WildScene': 'SIR2/WildSceneDataset'
}

In [4]:
# 2. Initialize the Evaluator
evaluator = ReflectionEvaluator(repo_dir='./Reflection_RemoVal_CVPR2024')

GPU 0 Free Memory: 10.40 GB
GPU 1 Free Memory: 9.89 GB
Models assigned to: cuda:0
Total number of paramerters in EfficientNet networks is 12233232 
Total number of requires_grad paramerters in EfficientNet networks is 12233232 


In [5]:
# 3. Run the loop and collect metrics
results_log = []

for name, rel_path in datasets_to_run.items():
    in_dir = f"{base_path}{rel_path}/blended"
    gt_dir = f"{base_path}{rel_path}/transmission_layer"
    
    psnr, ssim = evaluator.evaluate_dataset(
        dataset_name=name, 
        input_dir=in_dir, 
        gt_dir=gt_dir, 
        output_base_dir=output_path
    )
    
    results_log.append({'Dataset': name, 'PSNR': round(psnr, 4), 'SSIM': round(ssim, 4)})


Evaluating [nature20]: 20 images found.


nature20: 100%|██████████| 20/20 [00:08<00:00,  2.31it/s]



Evaluating [real20]: 20 images found.


real20: 100%|██████████| 20/20 [00:25<00:00,  1.26s/it]



Evaluating [OpenRR100]: 100 images found.


OpenRR100: 100%|██████████| 100/100 [01:42<00:00,  1.02s/it]



Evaluating [SIR2_Postcard]: 179 images found.


SIR2_Postcard: 100%|██████████| 179/179 [01:13<00:00,  2.43it/s]



Evaluating [SIR2_SolidObject]: 200 images found.


SIR2_SolidObject: 100%|██████████| 200/200 [01:17<00:00,  2.57it/s]



Evaluating [SIR2_WildScene]: 101 images found.


SIR2_WildScene: 100%|██████████| 101/101 [00:37<00:00,  2.72it/s]


In [6]:
# 4. Display the final summary table
df_results = pd.DataFrame(results_log)
print("\n=== FINAL BENCHMARK RESULTS ===")
display(df_results.T)


=== FINAL BENCHMARK RESULTS ===


,0,1,2,3,4,5
Dataset,nature20,real20,OpenRR100,SIR2_Postcard,SIR2_SolidObject,SIR2_WildScene
PSNR,26.0977,23.7379,25.6488,24.0898,26.6896,27.0049
SSIM,0.8437,0.8127,0.9333,0.8925,0.9262,0.9277


NVIDIA GeForce GTX 1080Ti.

In [8]:
%%bash
nvidia-smi

Sat Apr 18 15:36:12 2026       
+-----------------------------------------------------------------------------------------+
| NVIDIA-SMI 575.64.03              Driver Version: 575.64.03      CUDA Version: 12.9     |
|-----------------------------------------+------------------------+----------------------+
| GPU  Name                 Persistence-M | Bus-Id          Disp.A | Volatile Uncorr. ECC |
| Fan  Temp   Perf          Pwr:Usage/Cap |           Memory-Usage | GPU-Util  Compute M. |
|                                         |                        |               MIG M. |
|=========================================+========================+======================|
|   0  NVIDIA GeForce RTX 2080 Ti     Off |   00000000:05:00.0  On |                  N/A |
| 35%   34C    P8             15W /  260W |     334MiB /  11264MiB |      0%      Default |
|                                         |                        |                  N/A |
+-----------------------------------------+-----

---

## *Exp 1(b)*: Changing the interpolation technique and comparing the results

In [22]:
output_path = './results_mini/two_datasets_eval/'

In [23]:
datasets_to_run = {
    'OpenRR100': 'OpenRR100',
    'SIR2_Postcard': 'SIR2/PostcardDataset',
    'SIR2_SolidObject': 'SIR2/SolidObjectDataset',
    'SIR2_WildScene': 'SIR2/WildSceneDataset'
}

In [24]:
evaluator = ReflectionEvaluator(repo_dir='./Reflection_RemoVal_CVPR2024')

GPU 0 Free Memory: 10.38 GB
GPU 1 Free Memory: 10.19 GB
Models assigned to: cuda:0
Total number of paramerters in EfficientNet networks is 12233232 
Total number of requires_grad paramerters in EfficientNet networks is 12233232 


In [25]:
# 3. Run the loop and collect metrics
results_log = []

for name, rel_path in datasets_to_run.items():
    in_dir = f"{base_path}{rel_path}/blended"
    gt_dir = f"{base_path}{rel_path}/transmission_layer"
    
    psnr, ssim = evaluator.evaluate_dataset(
        dataset_name=name, 
        input_dir=in_dir, 
        gt_dir=gt_dir, 
        output_base_dir=output_path
    )
    
    results_log.append({'Dataset': name, 'PSNR': round(psnr, 4), 'SSIM': round(ssim, 4)})


Evaluating [OpenRR100]: 100 images found.


OpenRR100: 100%|██████████| 100/100 [01:42<00:00,  1.02s/it]



Evaluating [SIR2_Postcard]: 179 images found.


SIR2_Postcard: 100%|██████████| 179/179 [01:12<00:00,  2.47it/s]



Evaluating [SIR2_SolidObject]: 200 images found.


SIR2_SolidObject: 100%|██████████| 200/200 [01:17<00:00,  2.59it/s]



Evaluating [SIR2_WildScene]: 101 images found.


SIR2_WildScene: 100%|██████████| 101/101 [00:37<00:00,  2.68it/s]


=== FINAL BENCHMARK RESULTS ===
0	1	2	3	4	5
Dataset	nature20	real20	OpenRR100	SIR2_Postcard	SIR2_SolidObject	SIR2_WildScene
PSNR	26.0977	23.7379	25.6488	24.0898	26.6896	27.0049
SSIM	0.8437	0.8127	0.9333	0.8925	0.9262	0.9277

In [26]:
# 4. Display the final summary table
df_results = pd.DataFrame(results_log)
print("\n=== FINAL BENCHMARK RESULTS ===")
display(df_results.T)


=== FINAL BENCHMARK RESULTS ===


,0,1,2,3
Dataset,OpenRR100,SIR2_Postcard,SIR2_SolidObject,SIR2_WildScene
PSNR,25.6488,24.0898,26.6896,27.0049
SSIM,0.9333,0.8925,0.9262,0.9277
